server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java

```diff
@@ -87,7 +85,7 @@ public NodeInfo(StreamInput in) throws IOException {
         if (in.getTransportVersion().onOrAfter(TransportVersion.V_7_10_0)) {
             addInfoIfNonNull(AggregationInfo.class, in.readOptionalWriteable(AggregationInfo::new));
         }
-        if (in.getTransportVersion().onOrAfter(TRANSPORT_VERSION_ADVANCED_REMOTE_CLUSTER_SECURITY_CCS)) {
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
```

Comment:
This is one example where we have to keep the existing version number. Bumping it is not only an issue for tests but also for real rolling upgrade scenario. Because this logic is not guarded by the feature flag which means an older node (e.g. 8.9.0) already expects this field to be present. Bumping the version leads to incosistent behaviour between old and new node. This particular class here is critical because failing NodeInfo API prevents cluster to be formed ...

As discussed briefly in Slack, I have since revised the strategy for version bumping and *not* pushing for bumping everything. It will be more conservative to focus mainly on the classes that relate to entry points to a cluster. More concretely, here is a list of how updating will be done:

**MUST Update**
* NativeRolesStore - Role parsing depends on feature flag
* ApiKeyService (Filtering user roles for remote indices) - Same as above
* ApiKeyService (creating/updating API keys) - For both cross-cluster key check and remote indices filtering
* SecurityServerTransportInterceptor - Exit point to remote cluster
* CrossClusterAccessAuthenticationService - Entry point to remote cluster
* Authentication (maybeRewriteMetadataForApiKeyRoleDescriptors) - It filters out remote indices, so similar as above

**Update** (I think these are optional, but update feels safer)
* Authentication#doWriteTo - The check just throws error so no cross-wire concern
* Authentication#maybeRewriteForOlderVersion - As above
* RemoteClusterFeatureSetUsage - The overall `XPackUsageResponse` serialization is pretty lenient, so it does not really matter.
* Subject has a few assertions - Does not seem to be a practical issue

**MUST Keep**
* CCR action serialization - The field is known to old versions already and this is not affected by feature flag.
* SecurityFeatureSetUsage serialization - Similar to the above 
* RemoteConnectionInfo serialization- Part of SecurityFeatureSetUsage
* NodeInfo serialization - As described in the previous section

**Keep** (I think these are optional, but keep as is feels safer)
* RoleDescriptor serialization - Not affected by feature flag. It might not really happen in practice since we guard entries.
* GetUserPrivilegesResponse serialization - Does not cross wire
* PutRoleRequest serialization - Does not cross wire
* ApiKey serialization - Does not cross wire